### PR TITLE
created linked list class and associated test class

### PR DESCRIPTION
**Pull request by micahThor**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [] Picture of whiteboard

### CODE REVIEW

File: Data-Structures/src/test/java/linkedlist/LinkedListTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    @Test public void linkedList_testEmptyList() {
+
+        LinkedList<Integer> emptyIntegerList = new LinkedList<>();
+
+        int expected = 0;
+        int actual = emptyIntegerList.getSize();
+
+        assertEquals(expected, actual);
+    }
+
```
Comment (JamesDansie) [inline Data-Structures/src/test/java/linkedlist/LinkedListTest.java:18]: If there is setup you're repeating for each test you can use @Before to run before each test. For example;

LinkedList<Integer> intLinkedList = new LinkedList<>();
@Before 
public void setup(){
        intLinkedList.insert(22);
        intLinkedList.insert(8);
        intLinkedList.insert(54);
        intLinkedList.insert(7);
}

Will setup the linked list before each test. Not required at all, but might make life easier.
